providers/sadad/sadad.go

```diff
@@ -0,0 +1,138 @@
+package sadad
+
+import (
+	"context"
+	"crypto/aes"
+	b64 "encoding/base64"
+	"github.com/GoFarsi/paygap/client"
+	"github.com/GoFarsi/paygap/status"
+	"github.com/andreburgaud/crypt2go/ecb"
+	"google.golang.org/grpc/codes"
+	"math/rand"
+	"net/http"
+	"strings"
+	"time"
+)
+
+const (
+	baseUrl           = "https://sadad.shaparak.ir"
+	requestPaymentUrl = "/api/v0/Request/PaymentRequest"
+	requestVerifyUrl  = "/api/v0/Advice/Verify"
+)
+
+func New(client client.Transporter, terminalId string, amount int64,
```

Comment:
This parameters `amount int64, merchantKey string, returnUrl string, merchantId string, purchasePage string` is unrequited, New function is constructor for create sadad object.
Just get configuration and Auth details.

Reply:
Thanks.
Done.